src/components/pages/Invite/Invite.tsx

```diff
@@ -0,0 +1,158 @@
+import { BackHandler } from "react-native";
+import * as React from "react";
+import { Container } from "../../shared/elements";
+import { InviteCamera } from "./InviteCamera";
+import DropdownAlert from "react-native-dropdownalert";
+import { VideoPreview } from "../Camera/VideoPreview";
+import { MapStateToProps, connect } from "react-redux";
+import {
+  getLoggedInMember,
+  getInviteVideoRef
+} from "../../../store/selectors/authentication";
+import { Member } from "../../../store/reducers/members";
+import { RahaState } from "../../../store";
+import { generateToken } from "../../../helpers/token";
+
+enum InviteStep {
+  CAMERA,
+  VIDEO_PREVIEW,
+  SEND_INVITE
+}
+
+type ReduxStateProps = {
+  loggedInMember?: Member;
+};
+
+type OwnProps = {};
+
+type InviteProps = ReduxStateProps & OwnProps;
+
+type InviteState = {
+  step: InviteStep;
+  verifiedName?: string;
+  videoUri?: string;
+  videoDownloadUrl?: string;
+};
+
+export class InviteView extends React.Component<InviteProps, InviteState> {
+  inviteToken: string;
+  videoUploadRef: firebase.storage.Reference;
+  dropdown: any;
+
+  constructor(props: InviteProps) {
+    super(props);
+    this.inviteToken = generateToken();
+    this.videoUploadRef = getInviteVideoRef(this.inviteToken);
+    this.state = {
+      step: InviteStep.CAMERA
+    };
+  }
+
+  componentDidMount() {
+    BackHandler.addEventListener("hardwareBackPress", this._handleBackPress);
+  }
+
+  componentWillUnmount() {
+    BackHandler.removeEventListener("hardwareBackPress", this._handleBackPress);
+  }
+
+  _handleBackPress = () => {
+    const index = this.state.step.valueOf();
+    if (index === 0) {
+      // Exit out of Invite flow.
+      return false;
+    } else {
+      const previousStepKey = InviteStep[index - 1];
+      this.setState({
+        step: InviteStep[previousStepKey as keyof typeof InviteStep]
+      });
+      return true;
+    }
+  };
+
+  _verifyVideoUri = () => {
+    const videoUri = this.state.videoUri;
+    if (!videoUri) {
+      this.dropdown.alertWithType(
+        "error",
+        "Error: Can't show video",
+        "Invalid video. Please retake your video."
+      );
+      this.setState({
+        step: InviteStep.CAMERA
+      });
+    }
+    return videoUri;
+  };
+
+  _renderStep() {
+    switch (this.state.step) {
+      case InviteStep.CAMERA: {
+        return (
+          <InviteCamera
+            onVideoRecorded={(videoUri: string) => {
+              this.setState({
+                videoUri: videoUri,
+                step: InviteStep.VIDEO_PREVIEW
+              });
+            }}
+          />
+        );
+      }
+      case InviteStep.VIDEO_PREVIEW: {
+        const videoUri = this._verifyVideoUri();
+        if (!videoUri) {
+          return <React.Fragment />;
```

Comment:
Maybe log something as well? if this is unexpected

Reply:
Apparently logging can lag the app and in general I don't think we want console logs in our prod apps. I think we should log analytics when we have it. Otherwise I think the error dropdown should be enough to alert us during debugging.

Reply:
I just want at least something here so that it's obvious that this is a place we should be logging. I don't want to have to audit all the code to figure out all the places we might have missed that logs should occur; so at a minimum I think we should put a comment and even better would just to install that babel plugin and include it in our babelrc.

Reply:
Sure. I'll install the plugin in a follow up PR.